src/tree-sitter-node/NodeParserAdapter.ts

```diff
@@ -48,6 +50,9 @@ export class NodeParserAdapter implements ParserAdapter {
       case 'rust':
         this.parser.setLanguage(Rust)
         break
+      case 'go':
+        this.parser.setLanguage(Go)
+        break
       default:
         throw new Error(`Unsupported language: ${languageName}`)
```

Comment:
Consider handling the 'Unsupported language' error in a more user-friendly way. Instead of throwing an error, you could log a warning and fall back to a default language. This would make the application more robust. [medium]